from snakemake.utils import min_version

min_version("6.10.0")

# Configuration file containing all user-specified settings
configfile: "config/config.yml"

report: "report/workflow.rst"

include: "rules/import_qiime2_metadata.smk"
include: "rules/qiime2_process.smk"
include: "rules/dada2_qc.smk"
include: "rules/denovo_cluster.smk"
include: "rules/closedref_cluster.smk"
include: "rules/openref_cluster.smk"
include: "rules/seq_alignment.smk"
include: "rules/phylogeny_cluster.smk"
include: "rules/assign_taxonomy.smk"
include: "rules/convert_to_tsv.smk"
include: "rules/collapse_taxa.smk"
include: "rules/alpha_rarefaction.smk"
include: "rules/qiime2_core_metrics.smk"
include: "rules/rmd_report.smk"

rule all:
	input:
		"index.html",

		"qiime2_process/sample-metadata.qzv",


		"qiime2_process/demux.qza",


		"qiime2_process/demux.qzv",


		"qiime2_process/rep-seqs.qza",
		"qiime2_process/feature-table.qza",
		"qiime2_process/stats.qza",


		"qiime2_process/rep-seqs.qzv",
		"qiime2_process/feature-table.qzv",
		"qiime2_process/stats.qzv",


 		"qiime2_process/feature-table-dn-99.qza",
		"qiime2_process/rep-seqs-dn-99.qza",


		"qiime2_process/table-cr-85.qza",
		"qiime2_process/rep-seqs-cr-85.qza",
		"qiime2_process/unmatched-cr-85.qza",


		"qiime2_process/table-or-85.qza",
		"qiime2_process/rep-seqs-or-85.qza",
		"qiime2_process/new-ref-seqs-or-85.qza",

		"qiime2_process/aligned-rep-seqs.qza",
		"qiime2_process/masked-aligned-rep-seqs.qza",
		"qiime2_process/masked-aligned-rep-seqs.qzv",


		"qiime2_process/unrooted-tree.qza",
  		"qiime2_process/rooted-tree.qza",

		"qiime2_process/transformed/feature-table.tsv",
		"qiime2_process/transformed/taxonomy.tsv",

		"qiime2_process/feature-table-genus.qza",
		
		"qiime2_process/alpha-rarefaction.qzv",


		"qiime2_process/alpha_diversity/shannon_vector.qza",
		"qiime2_process/beta_diversity/bray_curtis_emperor.qzv"